addon/model.js

```diff
@@ -686,4 +689,29 @@ export class EmbeddedMegamorphicModel extends MegamorphicModel {
   toString() {
     return `<EmbeddedMegamorphicModel:${this.id}>`;
   }
+  serialize(options) {
+    return this._store.serializerFor('-ember-m3').serialize(new EmbeddedSnapshot(this), options);
+  }
+}
+
+export class EmbeddedSnapshot {
```

Comment:
Is this only needed in the non-custom-class world? If so, can we flag it?

Reply:
Needed also for the custom class world to wire up record saving for now